### PR TITLE
Date parsing

### PR DESCRIPTION
**Pull request by jh-bate**
@darinkrauss this resolves the issue that we were having with dates from the notes app - I have tested on devel with a simulator version of the notes app

So the reason that Lennart and I get this issue is we are in `+` UTC timezones to the `+13:00` wasn't escaped so it would drop that from the time and then when doing the conversion blow up. I know its an assumption that I have built in but let me know if its not acceptable as a band aid

### CODE REVIEW

**Comment by jh-bate** [comment]
[<0:ux 3:dev 0:tst> [bug] Research Unknown Error Occurred [Blip Notes]](https://trello.com/c/9Y2J1pNN/74-0-ux-3-dev-0-tst-bug-research-unknown-error-occurred-blip-notes)

File: lib/routes/utility.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+     * get the date as an ISO 8601 string from any valid datetime YYYY-MM-DDTHH:mm:ss.sssZ string
+     *
+     * @param {string} datetime
+     * @return {string} ISO 8601 string or null
+     */
+    getISODate: function(datetime) {
+      if (_.isEmpty(datetime)) {
+        return null;
+      }
+      datetime = datetime.replace(' ', '+');
```
Comment (darinkrauss) [inline lib/routes/utility.js]: What is this for?

File: test/unit/utility_tests.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    });
+    it('date, time and offset - UTC', function(done) {
+      var returnedDate = utility.getISODate('2016-12-22T16:27:10-08:00');
+      expect(returnedDate).to.exist;
+      expect(returnedDate).to.equal('2016-12-23T00:27:10.000Z');
+      done();
+    });
+  });
+
+});
```
Comment (darinkrauss) [inline test/unit/utility_tests.js]: Needs a newline at the end.

File: lib/routes/utility.js (hunk 1 above)
Comment (jh-bate) [inline lib/routes/utility.js]: This is what was happening in the case of UTC + timezones. Hence the bandaid solution

```
it('date, time and offset UTC', function(done) {
      var returnedDate = utility.getISODate('2016-12-22T16:27:10 13:00');
      expect(returnedDate).to.exist;
      expect(returnedDate).to.equal('2016-12-22T03:27:10.000Z');
      done();
    });
```

**Pull request merged**